homeassistant/components/sensor/fritzbox_callmonitor.py

```diff
@@ -30,10 +35,18 @@
 
 INTERVAL_RECONNECT = 60
 
+# Return cached results if phonebook was downloaded less then this time ago.
+MIN_TIME_PHONEBOOK_UPDATE = datetime.timedelta(hours=6)
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
     vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_PASSWORD, default='admin'): cv.string,
+    vol.Optional(CONF_USERNAME, default=''): cv.string,
```

Comment:
The reason the password was left as optional is that somebody might be concerned about hard-coding the router password into HA but might still want to use the call monitor (without phone book lookup). In the documentation it is mentioned that phone book support requires a username/password.

Concerning the default values, you're right! This should be interchanged. In fact this seems to be a copy & paste error as it is the same in the Fritz device tracker platform.